website_docs/instrumentation.md

```diff
@@ -17,30 +17,52 @@ ctx, span = tracer.Start(ctx, "helloWorld")
 defer span.End()
 ```
 
-In Go, the `context` package is used to store the active span. When you start a span, you'll get a handle on not only the span that's created, but the modified context that contains it. When starting a new span using this context, a parent-child relationship will automatically be created between the two spans, as seen here:
+In Go, the `context` package is used to store the active span. When you start a span, you'll get a handle on not only the span that's created, but the modified context that contains it.
+
+Once a span has completed, it is immutable and can no longer be modified.
+
+### Get the current span
+
+To get the current span, you'll need to pull it out of the `Context` you have a handle on:
+
+```go
+ctx := context.Background()
+span := trace.SpanFromContext(ctx)
+
+// Do something with the current span, optionally calling `span.End()` if you want it to end
+```
+
+This can helpful if you'd like to add information to the current span at a point in time.
+
+### Create nested spans
+
+You can create a nested span to track work in a nested operation.
+
+If the current `Context` you have a handle on already contains a span inside of it, creating a new span makes it a nested span. For example:
 
 ```go
 func parentFunction() {
+	// Create a span that tracks the work for `parentFunction()`
 	ctx := context.Background()
-	var parentSpan trace.Span
 	ctx, parentSpan = tracer.Start(ctx, "parent")
 	defer parentSpan.End()
-	// call our child function
+
+	// call the child function and start a nested span in there
 	childFunction(ctx)
-	// do more work, when this function ends, parentSpan will complete.
+
+	// do more work - when this function ends, parentSpan will complete.
 }
 
 func childFunction(ctx context.Context) {
-	var childSpan trace.Span
```

Comment:
same here, this becomes invalid code with this removal.

Reply:
I think it is fine, if somewhat unorthodox, to shadow `ctx` here since the incoming value can't be modified in any circumstance and any assignment into `ctx` is necessarily a new context distinct from the old:

```go
ctx, childSpan := tracer.Start(ctx, "child")
```